index.js

```diff
@@ -1,20 +1,21 @@
 /* global Blob */
 
-var once = require('once')
-
-module.exports = function getBlob (stream, mimeType, cb) {
-  if (typeof mimeType === 'function') return getBlob(stream, null, mimeType)
-  cb = once(cb)
-  var chunks = []
-  stream
-    .on('data', function (chunk) {
+module.exports = function (stream, mimeType) {
+  return new Promise(function (resolve, reject) {
+    if (typeof mimeType === 'function') {
+      reject(new Error('Invalid mimetype, expected string, got function.'))
```

Comment:
Rather than rejecting the promise, I'd rather have this function throw an exception immediately. Promise rejections are for runtime errors, thrown exceptions are for programmer errors (i.e. code that could never work).